### DISCUSSION TITLE
UTMStack v10, local install issue

### DISCUSSION THREAD

**Discussion by MaterialStep4** [utmstack/UTMStack#437] [Q&A] [answered]
My local install now won't load the UTMStack web portal.  I get a "502 Gateway Error" instead.

I tried an updated installer, but got this in /var/log/utm-setup.log:

Starting datasources_agent_manager ... error

ERROR: for datasources_agent_manager  Cannot start service datasources_agent_manager: driver failed programming external connectivity on endpoint datasources_agent_manager (27eb8c538e81a507359d5203241598cb09076c2a0ddaf8730637d82025f1e131): Error starting userland proxy: listen tcp4 0.0.0.0:9000: bind: address already in use

ERROR: for postgres  Cannot start service postgres: driver failed programming external connectivity on endpoint postgres (8b247f7a496e72fc2a1710ed4c980841368bf96c58c3d32ba9b10445505f895a): Error starting userland proxy: listen tcp4 0.0.0.0:5432: bind: address already in use

ERROR: for node1  Cannot start service node1: driver failed programming external connectivity on endpoint node1 (6f52a7284cfa1fcc15d36fff3ba1c12c6ebf68dc06c627151ba49c29a249f4b9): Error starting userland proxy: listen tcp4 0.0.0.0:9200: bind: address already in use

ERROR: for datasources_agent_manager  Cannot start service datasources_agent_manager: driver failed programming external connectivity on endpoint datasources_agent_manager (27eb8c538e81a507359d5203241598cb09076c2a0ddaf8730637d82025f1e131): Error starting userland proxy: listen tcp4 0.0.0.0:9000: bind: address already in use
Encountered errors while bringing up the project.

**Comment by osmontero** [comment]
Hello, this error is returned by Docker when a service tries to start using a port that is already in use by another service.

Please try using the command:
`docker stack rm utmstack`

Then restart the server and run the installer again.

**Comment by MaterialStep4** [comment]
Same result:

support@siem:~$ sudo docker stack rm utmstack
Removing service utmstack_agentmanager
Removing service utmstack_aws
Removing service utmstack_backend
Removing service utmstack_bitdefender
Removing service utmstack_correlation
Removing service utmstack_filebrowser
Removing service utmstack_frontend
Removing service utmstack_log-auth-proxy
Removing service utmstack_logstash
Removing service utmstack_mutate
Removing service utmstack_node1
Removing service utmstack_office365
Removing service utmstack_postgres
Removing service utmstack_socai
Removing service utmstack_sophos
Removing service utmstack_user-auditor
Removing network utmstack_default
support@siem:~$ sudo ./installer
### UTMStack Installer ###
Checking system requirements
Checking system requirements
Checking system requirements [OK]
Generating Stack configuration
Generating Stack configuration [OK]
Installing Stack. This may take a while.
Downloading  utmstack.azurecr.io/datasources:v10
v10: Pulling from datasources
Digest: sha256:a4b5d13cb439f959033b04022796f19f61a486c7edc46433132693a2cf26ddff
Status: Image is up to date for utmstack.azurecr.io/datasources:v10
utmstack.azurecr.io/datasources:v10
Downloading  ghcr.io/utmstack/utmstack/office365:v10
v10: Pulling from utmstack/utmstack/office365
Digest: sha256:b68dddba43d6940629a66afbbcde8fc11b5160c5237549f7d4991439d12547ae
Status: Image is up to date for ghcr.io/utmstack/utmstack/office365:v10
ghcr.io/utmstack/utmstack/office365:v10
Downloading  ghcr.io/utmstack/utmstack/bitdefender:v10
v10: Pulling from utmstack/utmstack/bitdefender
Digest: sha256:71a7e748a28910c29ef16bd3d861d274fb1a9efbc258f386b15b2478d8d34ca1
Status: Image is up to date for ghcr.io/utmstack/utmstack/bitdefender:v10
ghcr.io/utmstack/utmstack/bitdefender:v10
Downloading  ghcr.io/utmstack/filebrowser/filebrowser:v10
v10: Pulling from utmstack/filebrowser/filebrowser
Digest: sha256:f359cb6d52c8871b6c8578fcde728b76352a94f8ae1e02a5591287fec4a7aaa4
Status: Image is up to date for ghcr.io/utmstack/filebrowser/filebrowser:v10
ghcr.io/utmstack/filebrowser/filebrowser:v10
Downloading  ghcr.io/utmstack/utmstack/correlation:v10
v10: Pulling from utmstack/utmstack/correlation
Digest: sha256:47520e338f7eeb1caf679395182898ea3ca1c0e320de84302b9ced3782d82093
Status: Image is up to date for ghcr.io/utmstack/utmstack/correlation:v10
ghcr.io/utmstack/utmstack/correlation:v10
Downloading  ghcr.io/utmstack/utmstack/log-auth-proxy:v10
v10: Pulling from utmstack/utmstack/log-auth-proxy
Digest: sha256:3c28ff372268b1f3fa545f0b52ac7f08076512a4b6ebfee592727c049b59deae
Status: Image is up to date for ghcr.io/utmstack/utmstack/log-auth-proxy:v10
ghcr.io/utmstack/utmstack/log-auth-proxy:v10
Downloading  ghcr.io/utmstack/utmstack/mutate:v10
v10: Pulling from utmstack/utmstack/mutate
Digest: sha256:d5be51758ea494bb8016d7af2db680b199b3f1a780afbc559a6ac447719c0f08
Status: Image is up to date for ghcr.io/utmstack/utmstack/mutate:v10
ghcr.io/utmstack/utmstack/mutate:v10
Downloading  ghcr.io/utmstack/utmstack/agent-manager:v10
v10: Pulling from utmstack/utmstack/agent-manager
Digest: sha256:efa14467463a7aa5d7b8e31940968d503b2c1a275a69684ad5ab435473e83c2a
Status: Image is up to date for ghcr.io/utmstack/utmstack/agent-manager:v10
ghcr.io/utmstack/utmstack/agent-manager:v10
Downloading  utmstack.azurecr.io/postgres:v10
v10: Pulling from postgres
Digest: sha256:0bab94fe5f78915c579d825ab43cbb08153bc72efad70445584d62a5396815a6
Status: Image is up to date for utmstack.azurecr.io/postgres:v10
utmstack.azurecr.io/postgres:v10
Downloading  ghcr.io/utmstack/utmstack/backend:v10
v10: Pulling from utmstack/utmstack/backend
Digest: sha256:9a1c3a465f3d66a749978f3ccfbfe499042c059a5bcfe1a111d0c943e411ef47
Status: Image is up to date for ghcr.io/utmstack/utmstack/backend:v10
ghcr.io/utmstack/utmstack/backend:v10
Downloading  utmstack.azurecr.io/opensearch:v10
v10: Pulling from opensearch
Digest: sha256:3b9a729d9c3c8c9d37bb03748103aa514793ebf9b4db818b381c84ba176b154f
Status: Image is up to date for utmstack.azurecr.io/opensearch:v10
utmstack.azurecr.io/opensearch:v10
Downloading  utmstack.azurecr.io/logstash:v10
v10: Pulling from logstash
Digest: sha256:c9392f17a7dcc1ef1d2e95ede62697d4aedc9f4f6b2f122f2dbd74bb729fe27b
Status: Image is up to date for utmstack.azurecr.io/logstash:v10
utmstack.azurecr.io/logstash:v10
Downloading  ghcr.io/utmstack/utmstack/frontend:v10
v10: Pulling from utmstack/utmstack/frontend
Digest: sha256:6352342bf2e6d4eba53484b9086ac3e63e39e1786b445bef8e4e577291d911e2
Status: Image is up to date for ghcr.io/utmstack/utmstack/frontend:v10
ghcr.io/utmstack/utmstack/frontend:v10
Downloading  utmstack.azurecr.io/datasources:v10
v10: Pulling from datasources
Digest: sha256:a4b5d13cb439f959033b04022796f19f61a486c7edc46433132693a2cf26ddff
Status: Image is up to date for utmstack.azurecr.io/datasources:v10
utmstack.azurecr.io/datasources:v10
Downloading  ghcr.io/utmstack/soc-ai/soc-ai:v10
v10: Pulling from utmstack/soc-ai/soc-ai
Digest: sha256:4b5462291f9dec1d1f4b1483db62be95a2eff361fd05a4cbfd8107c50c084259
Status: Image is up to date for ghcr.io/utmstack/soc-ai/soc-ai:v10
ghcr.io/utmstack/soc-ai/soc-ai:v10
Downloading  ghcr.io/utmstack/utmstack/user-auditor:v10
v10: Pulling from utmstack/utmstack/user-auditor
Digest: sha256:c0aa228ffc0cec2198006cee73a5542ecb234a6740e63b2252875e7c7f020abe
Status: Image is up to date for ghcr.io/utmstack/utmstack/user-auditor:v10
ghcr.io/utmstack/utmstack/user-auditor:v10
Creating network utmstack_default
Creating service utmstack_user-auditor
Creating service utmstack_mutate
Creating service utmstack_backend
Creating service utmstack_postgres
Creating service utmstack_correlation
Creating service utmstack_filebrowser
Creating service utmstack_aws
Creating service utmstack_sophos
Creating service utmstack_bitdefender
Creating service utmstack_log-auth-proxy
Creating service utmstack_agentmanager
Creating service utmstack_socai
Creating service utmstack_office365
Creating service utmstack_frontend
Creating service utmstack_node1
Creating service utmstack_logstash
Installing Stack [OK]
Installing reverse proxy. This may take a while.
Hit:1 https://download.docker.com/linux/ubuntu jammy InRelease
Hit:2 http://us.archive.ubuntu.com/ubuntu jammy InRelease
Get:3 http://us.archive.ubuntu.com/ubuntu jammy-updates InRelease [119 kB]
Hit:4 http://us.archive.ubuntu.com/ubuntu jammy-backports InRelease
Get:5 http://us.archive.ubuntu.com/ubuntu jammy-security InRelease [110 kB]
Get:6 https://ppa.launchpadcontent.net/oisf/suricata-stable/ubuntu jammy InRelease [17.5 kB]
Get:7 http://us.archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [1,366 kB]
Get:8 https://ppa.launchpadcontent.net/oisf/suricata-stable/ubuntu jammy/main amd64 Packages [1,376 B]
Get:9 http://us.archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1,043 kB]
Fetched 2,658 kB in 1s (1,792 kB/s)
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
3 packages can be upgraded. Run 'apt list --upgradable' to see them.
W: https://download.docker.com/linux/ubuntu/dists/jammy/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
nginx is already the newest version (1.18.0-6ubuntu14.4).
The following packages were automatically installed and are no longer required:
  libflashrom1 libftdi1-2
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 3 not upgraded.
Installing reverse proxy [OK]
Waiting for Backend to be ready. This may take a while.
**backend start failure: <nil>**

**Reply by jdieguez89** [reply]
@MaterialStep4 can you please provide the logs after "backend start failure:" line??

**Comment by MaterialStep4** [comment]
Tried again, with a reboot, then ran the installer.  It still failed.

**Comment by MaterialStep4** [comment]
ERROR: for node1  Cannot start service node1: driver failed programming external connectivity on endpoint node1 (68994e992d8a812dafd2a8a7a54e18844dea6c0c97fcbdd6e66bc226c2543694): Error starting userland proxy: listen tcp4 0.0.0.0:9200: bind: address already in use
Encountered errors while bringing up the project.
Found multiple config files with supported names: docker-compose.yml, compose.yml
Using docker-compose.yml

The Docker Engine you're using is running in swarm mode.

Compose does not use swarm mode to deploy services to multiple nodes in a swarm. All containers will be scheduled on the current node.

To deploy your application across the swarm, use `docker stack deploy`.

Starting postgres ...
Starting node1    ...
openvas is up-to-date
wazuh is up-to-date
autoupdate is up-to-date
watchtower is up-to-date
datasources_transporter is up-to-date
datasources_probe_api is up-to-date
datasources_mutate is up-to-date
filebrowser is up-to-date
Starting datasources_agent_manager ...
logstash is up-to-date
Host is already in use by another container
Starting postgres                  ... error

ERROR: for postgres  Cannot start service postgres: driver failed programming external connectivity on endpoint postgres (877ea6e26d45289fb17252f942984a9dcc0463ab495f94bb163cb723e807c472): Error starting userland proxy: listen tcp4 0.0.0.0:5432: bind: address already in use
Host is already in use by another container
Starting node1                     ... error

ERROR: for node1  Cannot start service node1: driver failed programming external connectivity on endpoint node1 (fac941b5340b4103347aeaf3907aef8f867a21374b80823ae98b97c04475848f): Error starting userland proxy: listen tcp4 0.0.0.0:9200: bind: address already in use
Host is already in use by another container
Starting datasources_agent_manager ... error

ERROR: for datasources_agent_manager  Cannot start service datasources_agent_manager: driver failed programming external connectivity on endpoint datasources_agent_manager (98ed6d801d9a549f6a2ea86bfd720a1c758148495fe510362c3b451089caf47c): Error starting userland proxy: listen tcp4 0.0.0.0:9000: bind: address already in use

ERROR: for postgres  Cannot start service postgres: driver failed programming external connectivity on endpoint postgres (877ea6e26d45289fb17252f942984a9dcc0463ab495f94bb163cb723e807c472): Error starting userland proxy: listen tcp4 0.0.0.0:5432: bind: address already in use

ERROR: for node1  Cannot start service node1: driver failed programming external connectivity on endpoint node1 (fac941b5340b4103347aeaf3907aef8f867a21374b80823ae98b97c04475848f): Error starting userland proxy: listen tcp4 0.0.0.0:9200: bind: address already in use

ERROR: for datasources_agent_manager  Cannot start service datasources_agent_manager: driver failed programming external connectivity on endpoint datasources_agent_manager (98ed6d801d9a549f6a2ea86bfd720a1c758148495fe510362c3b451089caf47c): Error starting userland proxy: listen tcp4 0.0.0.0:9000: bind: address already in use
Encountered errors while bringing up the project.
Found multiple config files with supported names: docker-compose.yml, compose.yml
Using docker-compose.yml

The Docker Engine you're using is running in swarm mode.

Compose does not use swarm mode to deploy services to multiple nodes in a swarm. All containers will be scheduled on the current node.

To deploy your application across the swarm, use `docker stack deploy`.

datasources_mutate is up-to-date
Starting postgres ...
Starting node1    ...
openvas is up-to-date
wazuh is up-to-date
autoupdate is up-to-date
watchtower is up-to-date
filebrowser is up-to-date
datasources_transporter is up-to-date
datasources_probe_api is up-to-date
logstash is up-to-date
Starting datasources_agent_manager ...
Host is already in use by another container
Starting node1                     ... error

ERROR: for node1  Cannot start service node1: driver failed programming external connectivity on endpoint node1 (1da12a79b11ccb5a2f73ed17f208b1a6792d6c8bbb9fb859ea799a95dcc04aca): Error starting userland proxy: listen tcp4 0.0.0.0:9200: bind: address already in use
Host is already in use by another container
Starting postgres                  ... error

ERROR: for postgres  Cannot start service postgres: driver failed programming external connectivity on endpoint postgres (5399ee165cb82319fb2fd82fb17b5626b9e04efbf7e145d6ec1cf2ab29b70597): Error starting userland proxy: listen tcp4 0.0.0.0:5432: bind: address already in use
Host is already in use by another container
Starting datasources_agent_manager ... error

ERROR: for datasources_agent_manager  Cannot start service datasources_agent_manager: driver failed programming external connectivity on endpoint datasources_agent_manager (2981bb1cef10df0fc5d5f754f622ee04f837462753d4e96610ef59d23b264788): Error starting userland proxy: listen tcp4 0.0.0.0:9000: bind: address already in use

ERROR: for node1  Cannot start service node1: driver failed programming external connectivity on endpoint node1 (1da12a79b11ccb5a2f73ed17f208b1a6792d6c8bbb9fb859ea799a95dcc04aca): Error starting userland proxy: listen tcp4 0.0.0.0:9200: bind: address already in use

ERROR: for postgres  Cannot start service postgres: driver failed programming external connectivity on endpoint postgres (5399ee165cb82319fb2fd82fb17b5626b9e04efbf7e145d6ec1cf2ab29b70597): Error starting userland proxy: listen tcp4 0.0.0.0:5432: bind: address already in use

ERROR: for datasources_agent_manager  Cannot start service datasources_agent_manager: driver failed programming external connectivity on endpoint datasources_agent_manager (2981bb1cef10df0fc5d5f754f622ee04f837462753d4e96610ef59d23b264788): Error starting userland proxy: listen tcp4 0.0.0.0:9000: bind: address already in use
Encountered errors while bringing up the project.
Found multiple config files with supported names: docker-compose.yml, compose.yml
Using docker-compose.yml

The Docker Engine you're using is running in swarm mode.

Compose does not use swarm mode to deploy services to multiple nodes in a swarm. All containers will be scheduled on the current node.

To deploy your application across the swarm, use `docker stack deploy`.

Starting postgres ...
Starting node1    ...
openvas is up-to-date
wazuh is up-to-date
autoupdate is up-to-date
watchtower is up-to-date
filebrowser is up-to-date
datasources_transporter is up-to-date
datasources_probe_api is up-to-date
datasources_mutate is up-to-date
Starting datasources_agent_manager ...
logstash is up-to-date
Host is already in use by another container
Host is already in use by another container
Starting node1                     ... error

ERROR: for node1  Cannot start service node1: driver failed programming external connectivity on endpoint node1 (2b2b6ea21ef88c56af332d0837e30cbaa8719a9d782a1b3dd4c187b86fc9860a): Error starting userland proxy: listen tcp4 0.0.0.0:9200: bind: address already in use
Starting postgres                  ... error

ERROR: for postgres  Cannot start service postgres: driver failed programming external connectivity on endpoint postgres (de43d31d431b8b51b9dbcbccbd0d97b9f197f3c074a860a76860d2f63861a30f): Error starting userland proxy: listen tcp4 0.0.0.0:5432: bind: address already in use
Host is already in use by another container
Starting datasources_agent_manager ... error

ERROR: for datasources_agent_manager  Cannot start service datasources_agent_manager: driver failed programming external connectivity on endpoint datasources_agent_manager (98f3903ff896f16c3299a47c987e7ef25f6f9c152624859af9fbad8340f5d928): Error starting userland proxy: listen tcp4 0.0.0.0:9000: bind: address already in use

ERROR: for node1  Cannot start service node1: driver failed programming external connectivity on endpoint node1 (2b2b6ea21ef88c56af332d0837e30cbaa8719a9d782a1b3dd4c187b86fc9860a): Error starting userland proxy: listen tcp4 0.0.0.0:9200: bind: address already in use

ERROR: for postgres  Cannot start service postgres: driver failed programming external connectivity on endpoint postgres (de43d31d431b8b51b9dbcbccbd0d97b9f197f3c074a860a76860d2f63861a30f): Error starting userland proxy: listen tcp4 0.0.0.0:5432: bind: address already in use

ERROR: for datasources_agent_manager  Cannot start service datasources_agent_manager: driver failed programming external connectivity on endpoint datasources_agent_manager (98f3903ff896f16c3299a47c987e7ef25f6f9c152624859af9fbad8340f5d928): Error starting userland proxy: listen tcp4 0.0.0.0:9000: bind: address already in use
Encountered errors while bringing up the project.
Found multiple config files with supported names: docker-compose.yml, compose.yml
Using docker-compose.yml

The Docker Engine you're using is running in swarm mode.

Compose does not use swarm mode to deploy services to multiple nodes in a swarm. All containers will be scheduled on the current node.

To deploy your application across the swarm, use `docker stack deploy`.

Starting postgres ...
Starting node1    ...
openvas is up-to-date
wazuh is up-to-date
autoupdate is up-to-date
watchtower is up-to-date
filebrowser is up-to-date
datasources_transporter is up-to-date
datasources_probe_api is up-to-date
datasources_mutate is up-to-date
Starting datasources_agent_manager ...
logstash is up-to-date
Host is already in use by another container
Starting postgres                  ... error

ERROR: for postgres  Cannot start service postgres: driver failed programming external connectivity on endpoint postgres (8b247f7a496e72fc2a1710ed4c980841368bf96c58c3d32ba9b10445505f895a): Error starting userland proxy: listen tcp4 0.0.0.0:5432: bind: address already in use
Host is already in use by another container
Starting node1                     ... error

ERROR: for node1  Cannot start service node1: driver failed programming external connectivity on endpoint node1 (6f52a7284cfa1fcc15d36fff3ba1c12c6ebf68dc06c627151ba49c29a249f4b9): Error starting userland proxy: listen tcp4 0.0.0.0:9200: bind: address already in use
Host is already in use by another container
Starting datasources_agent_manager ... error

ERROR: for datasources_agent_manager  Cannot start service datasources_agent_manager: driver failed programming external connectivity on endpoint datasources_agent_manager (27eb8c538e81a507359d5203241598cb09076c2a0ddaf8730637d82025f1e131): Error starting userland proxy: listen tcp4 0.0.0.0:9000: bind: address already in use

ERROR: for postgres  Cannot start service postgres: driver failed programming external connectivity on endpoint postgres (8b247f7a496e72fc2a1710ed4c980841368bf96c58c3d32ba9b10445505f895a): Error starting userland proxy: listen tcp4 0.0.0.0:5432: bind: address already in use

ERROR: for node1  Cannot start service node1: driver failed programming external connectivity on endpoint node1 (6f52a7284cfa1fcc15d36fff3ba1c12c6ebf68dc06c627151ba49c29a249f4b9): Error starting userland proxy: listen tcp4 0.0.0.0:9200: bind: address already in use

ERROR: for datasources_agent_manager  Cannot start service datasources_agent_manager: driver failed programming external connectivity on endpoint datasources_agent_manager (27eb8c538e81a507359d5203241598cb09076c2a0ddaf8730637d82025f1e131): Error starting userland proxy: listen tcp4 0.0.0.0:9000: bind: address already in use
Encountered errors while bringing up the project.

_________________________________________________
Jeff Slauson – IT Consultant – Creative Software Services, Inc.
***@***.******@***.***> – C: 319.310.0256 – O: 319.362.7722 ex.114

**Reply by c3s4rfred** [reply, marked as answer]
Hi, @MaterialStep4, according to the logs that you provided, maybe you have installed UTMStack  on a "non clean" system, we recommend you to install it from scratch on a new system according to our guides. Use the latest version. Also, check your network, some required ports needs to be open so UTMStack can work as expected.

**Comment by MaterialStep4** [comment]
I did a full format and reinstall of UTMStack, and the issue is resolved.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
